### PR TITLE
support partial cache deleting

### PR DESCRIPTION
**Pull request by dana-resonai**

### CODE REVIEW

File: yabt/gs_global_cache.py [hunk 1]
```diff
@@ -50,21 +50,31 @@ def has_cache(self, target_hash: str):
...
                                      SUMMARY_FILE)).exists()
 
-    def download_summary(self, target_hash: str, dst: str):
-        self.bucket.blob(join(self.targets_dir, target_hash,
-                              SUMMARY_FILE)).download_to_filename(dst)
-
-    def download_artifacts_meta(self, target_hash: str, dst: str):
-        self.bucket.blob(join(self.targets_dir, target_hash,
-                              ARTIFACTS_FILE)).download_to_filename(dst)
+    def download_summary(self, target_hash: str, dst: str) -> bool:
```
Comment (dlehavi-resonai) [inline yabt/gs_global_cache.py]: please unite the two functions and put the file as a parameter

Comment (dlehavi-resonai) [reply yabt/gs_global_cache.py]: not on the interface level

File: yabt/fs_global_cache.py [hunk 2]
```diff
@@ -43,18 +43,28 @@ def __init__(self, directory='/tmp/cache'):
     def has_cache(self, target_hash: str):
         return isdir(join(self.targets_dir, target_hash))
 
-    def download_summary(self, target_hash: str, dst: str):
+    def download_summary(self, target_hash: str, dst: str) -> bool:
```
Comment (dlehavi-resonai) [inline yabt/fs_global_cache.py]: now that you support "random erasure", you can add a test which:
- erases 10% of summaries
- erases 10% of artifacts
and still the cache magically downloads what it's supposed to.

File: yabt/gs_global_cache.py [hunk 3]
```diff
@@ -59,24 +59,30 @@ def has_cache(self, target_hash: str):
...
+    def download_artifacts_meta(self, target_hash: str, dst: str) -> bool:
+        return self.download_meta_file(target_hash, ARTIFACTS_FILE, dst)
 
-    def download_artifacts_meta(self, target_hash: str, dst: str):
+    def download_meta_file(self, target_hash: str, src: str, dst: str) -> bool:
         self._create_client()
-        self.bucket.blob(join(self.targets_dir, target_hash,
-                              ARTIFACTS_FILE)).download_to_filename(dst)
+        src_blob = self.bucket.blob(join(self.targets_dir, target_hash, src))
+        if not src_blob.exists():
```
Comment (eyal-resonai) [inline yabt/gs_global_cache.py]: this test cost time and network. it is better to do
```
try:
  src_blob.download_to_filename(dst)
except gcloud.exceptions.NotFound:
  return False
return True
```

File: yabt/gs_global_cache.py [hunk 4]
```diff
@@ -59,24 +59,30 @@ def has_cache(self, target_hash: str):
...
         self._create_client()
         if artifacts_hashes:
             # TODO(Dana): make this work in batch.
             # see https://github.com/googleapis/google-cloud-python/issues/3139
             for artifact_hash in artifacts_hashes:
-                self.bucket.blob(join(self.artifacts_dir, artifact_hash))\
-                    .download_to_filename(join(dst, artifact_hash))
+                src_blob = self.bucket.blob(join(self.artifacts_dir,
+                                                 artifact_hash))
+                if src_blob.exists():
```
Comment (eyal-resonai) [inline yabt/gs_global_cache.py]: same here

**Pull request merged**